src/monitor/Monitor.ts

```diff
@@ -902,16 +902,16 @@ export class Monitor {
     // provided) to fetch for latest events.
     // Else, if running in loop mode (pollingDelay != 0), start with the latest block and on next loops continue from
     // where the last one ended.
-    const latestBlockNumber = (await provider.getBlock("latest")).number;
+    const latestBlockSearched = (await provider.getBlock("latest")).number;
```

Comment:
This one looks incorrect since it is actually referencing the head of the chain.

```suggestion
    const latestBlockNumber = (await provider.getBlock("latest")).number;
```